frontend/src/components/labwork/step/LabworkStep.tsx

```diff
@@ -43,55 +43,64 @@ const LabworkStep = ({ protocol, step, stepSamples }: LabworkStepPageProps) => {
 		}
 	}, [stepSamples])
 
-	function handleSetFilter(filterKey: string, value: FilterValue, description: FilterDescription) {
-		if(typeof description === 'undefined') {
-			return
-		}
-		dispatch(setFilter(step.id, description, value))
-	}
+	const handleSetFilter = useCallback(
```

Comment:
I'm not convinced by all these additional useCallback. My understanding is you save during rendering by reusing the function but i don't really see any gain to use it on such small functions. As far as i can see you cannot reuse the values or the results so in the end you are most likely just adding test to the function with a risk of having the wrong function if the cache is not refreshed when it should. The fact that you do not set dependencies will likely lead to the wrong values being used in the function.

Reply:
You're absolutely right about the dependencies - there were some that I needed to add. I added the react-hooks linter plugin to check for missing dependencies, to avoid that in the future, and I added all the missing dependencies.

I do want to continue using these hooks and other react best practices though, and to consider rendering performance a bit more.